buda/entities/signin.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package buda;
+
+import "buda/entities/uuid.proto";
+import "google/protobuf/timestamp.proto";
+import "buda/entities/product.proto";
+
+message Signin {
+    enum SigninOption {
+      EMAIL = 0;
+      FACEBOOK = 1;
+      TWITTER = 2;
+      LINKEDIN = 3;
+      GOOGLE = 4;
+    }
+
+    enum SigninClient {
+        WEB = 0;
+        PUBLISH = 1;
+        ANALYZE = 3;
+        IPHONE = 4;
+        ANDROID = 5;
+        OTHER = 6;
+    }
+
+
+    message UserAgent {
+        string browser = 1;
+        string version = 2;
+        string os = 3;
+        string device = 4;
+      }
+
+    Uuid id = 1;
+    Uuid user_id = 2;
```

Comment:
What `user_id` is this? The `user_id` for the given product shared below in the `Product product = 7` field?

I ask b/c we're moving towards global account ids in sign on. Roughly the schema of the Account model used for authentication is something like this:
```js
{
  _id: ObjectId, // the global Buffer Account id - This will be a string in the Protobuf message
  productlinks: [ // a list of foreign keys to users in each product's database
    {
      productName: String, // ex. "publish," "reply," "analyze,"
      foreignKey: String,  // ex. "52bc7a190a0e32ec4c000020" (my Publish user id), 
                           // ex. "WCzMd_XRawABcQ9M" (my Reply user id)
    }
  ]
}
```

Reply:
Maybe we could have two fields instead?

* `String account_id` for the global Account model's `_id` value
* `String product_user_id` OR (`String publish_user_id` `String reply_user_id`) - This could be deprecated in the future once all of our models reference the account id